src/OrbitGl/TrackManager.cpp

```diff
@@ -464,6 +463,15 @@ ThreadTrack* TrackManager::GetOrCreateThreadTrack(uint32_t tid) {
   return track.get();
 }
 
+std::optional<ThreadTrack*> TrackManager::GetThreadTrack(uint32_t tid) {
+  std::lock_guard<std::recursive_mutex> lock(mutex_);
+  std::shared_ptr<ThreadTrack> track = thread_tracks_[tid];
```

Comment:
This is not doing what you want because `thread_tracks_[tid]` would default construct a shared_ptr in `thread_tracks_` if there is none with that `tid`.

You can use `thread_tracks_.contains` or `thread_tracks_.find` to do what you want.